cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -412,38 +390,81 @@ def test_cloud_provision_from_template_with_attached_disks(
     if provider.one_of(AzureProvider):
         recursive_update(inst_args, {'environment': {'availability_zone': provisioning("av_set")}})
 
-    device_name = "/dev/sd{}"
+    device_name = "vd{}"
     device_mapping = []
 
-    with provider.mgmt.with_volumes(1, n=disks) as volumes:
-        for i, volume in enumerate(volumes):
-            device_mapping.append((volume, device_name.format(chr(ord("b") + i))))
-        # Set up automate
+    volumes = provider.mgmt.volume_configurations(1, n=disks)
+
+    @request.addfinalizer
+    def delete_volumes():
+        for volume in volumes:
+            provider.mgmt.delete_volume(volume)
+
+    # Set up automate
+    for i, volume in enumerate(volumes, 0):
+        if i == 0:
```

Comment:
I think the questions in review in this block warrant more in-line comments of the intent here. Why boot_index is `-1` on the non-bootable disk(s), that the device_name 'counts' through 'vda', 'vdb', etc.

Reply:
ok